scheduler/core/events/cycle/cycle.py

```diff
@@ -240,8 +226,22 @@ def _handle_updates(self,
                     nightly_timeline
                 )
 
-            # Get a new selection and request a new plan if the night is not done
-            if not update.done:
+            # If the night is done, get a final plan and add it to the timeline.
+            # Otherwise, get a new selection and request a new plan
+            if update.done:
+                _logger.debug('Night done. Wrapping up final plan')
+                final_plan = nightly_timeline.get_final_plan(NightIndex(night_idx),
+                                                             site,
+                                                             self.change_monitor.is_site_unblocked(site))
```

Comment:
I guess this would force to get the final plan even if the site is blocked. Let me test this before I can approve it.